### 초기설정

In [1]:
%%capture
!pip install koreanize-matplotlib
import koreanize_matplotlib

In [2]:
import pandas as pd

# df_2018 = pd.read_excel('/content/2018.xlsx')
# df_2019 = pd.read_excel('/content/2019.xlsx')
# df_2020 = pd.read_excel('/content/2020.xlsx')
# df_2021 = pd.read_excel('/content/2021.xlsx')
df_2022 = pd.read_excel('/content/2022.xlsx')
df_2023 = pd.read_excel('/content/2023.xlsx')
df_2024 = pd.read_excel('/content/2024.xlsx')

In [3]:
# grade_map = {
#     '① 전혀 그렇지 않다': 1,
#     '② 별로 그렇지 않다': 2,
#     '③ 보통이다': 3,
#     '④ 대체로 그렇다': 4,
#     '⑤ 매우 그렇다': 5
# }

# df_2018['Q16'] = df_2018['Q16'].map(grade_map)

In [4]:
# df_2018['국가명'] = df_2018['COUNTRY']

In [5]:
country_map = {
    1: '일본',
    2: '중국',
    3: '홍콩',
    4: '싱가포르',
    5: '대만',
    6: '태국',
    7: '말레이시아',
    8: '호주',
    9: '미국',
    10: '캐나다',
    11: '영국',
    12: '독일',
    13: '프랑스',
    14: '러시아',
    15: '중동',
    16: '인도',
    17: '필리핀',
    18: '인도네시아',
    19: '베트남',
    20: '몽골',
    97: '기타'
}

# 2018은 이미 COUNTY 컬럼에 국가명이 적혀있음.
# df_2019['국가명'] = df_2019['D_COU1'].map(country_map)
# df_2020['국가명'] = df_2020['D_NAT'].map(country_map)
# df_2021['국가명'] = df_2021['D_NAT'].map(country_map)
df_2022['국가명'] = df_2022['D_NAT'].map(country_map)
df_2023['국가명'] = df_2023['D_NAT'].map(country_map)
df_2024['국가명'] = df_2024['D_NAT'].map(country_map)

- 활동명 매핑 (주요활동/만족활동) - 2022~2024

In [6]:
# 1. 활동명 매핑
q4a_mapping_2022 = {
    "Q4A01": "식도락 관광", "Q4A02": "쇼핑", "Q4A03": "자연경관 감상", "Q4A04": "고궁/역사 유적지 방문",
    "Q4A05": "전통문화체험", "Q4A06": "박물관, 전시관 관람", "Q4A07": "K-POP/한류스타 공연장 방문",
    "Q4A08": "드라마/영화 촬영지 방문", "Q4A09": "연극, 뮤지컬, 발레 등 공연 관람", "Q4A10": "지역 축제 참여",
    "Q4A11": "유흥/오락", "Q4A12": "뷰티/의료관광", "Q4A13": "스포츠/레포츠 및 비즈니스 활동",
    "Q4A14": "업무수행", "Q4A15": "연수/교육/연구", "Q4A16": "시찰(산업시설 등)", "Q4A17": "기타"
}

q4b_code_mapping_2022 = {
    1: "식도락 관광", 2: "쇼핑", 3: "자연경관 감상", 4: "고궁/역사 유적지 방문",
    5: "전통문화체험", 6: "박물관, 전시관 관람", 7: "K-POP/한류스타 공연장 방문",
    8: "드라마/영화 촬영지 방문", 9: "연극, 뮤지컬, 발레 등 공연 관람", 10: "지역 축제 참여",
    11: "유흥/오락", 12: "뷰티/의료관광", 13: "스포츠/레포츠 및 비즈니스 활동",
    14: "업무수행", 15: "연수/교육/연구", 16: "시찰(산업시설 등)", 17: "기타"
}

q4a_cols = list(q4a_mapping_2022.keys())
q4b_cols = ["Q4B1", "Q4B2", "Q4B3"]

# 2. 주요활동 추출 함수
def get_q4a_activities(row):
    return [q4a_mapping_2022[col] for col in q4a_cols if col in row and pd.notna(row[col])]

# 3. 만족활동 추출 함수
def get_q4b_satisfaction(row):
    return [q4b_code_mapping_2022.get(int(row[col]), f"코드 {row[col]}")
            for col in q4b_cols if col in row and pd.notna(row[col])]

# 4. 적용 (df_2022가 정의된 상태에서만)
df_2022["주요활동"] = df_2022.apply(get_q4a_activities, axis=1)
df_2022["만족활동"] = df_2022.apply(get_q4b_satisfaction, axis=1)


In [7]:
# 1. 활동명 매핑
q8a_mapping_2023 = {
    "Q8a01": "식도락(음식/미식) 관광", "Q8a02": "쇼핑", "Q8a03": "자연경관 감상", "Q8a04": "고궁/역사 유적지 방문",
    "Q8a05": "전통문화체험 (한복체험, 태권도, 한국음식 만들기 등)", "Q8a06": "박물관, 전시관 관람",
    "Q8a07": "K-POP/한류스타 공연장 및 드라마/영화 촬영지 방문", "Q8a08": "연극, 뮤지컬, 발레 등 공연 관람",
    "Q8a09": "지역 축제 참여", "Q8a10": "유흥(나이트라이프 활동 및 카지노 등)", "Q8a11": "오락(놀이공원 및 테마파크 방문 등)",
    "Q8a12": "뷰티/미용 관광(마사지샵, 헤어샵, 네일케어 등 방문)", "Q8a13": "치료/건강검진(피부과, 성형외과, 내외과 등 입원, 수술 등)",
    "Q8a14": "스포츠/레포츠 관람 (야구, 축구, 농구 등)", "Q8a15": "스포츠/레포츠 참가 (스키, 수영, 골프, 캠핑 등)",
    "Q8a16": "국제회의(컨벤션, 국제회의, 전시 등)", "Q8a17": "비즈니스 전문활동(업무회의, 산업시설 시찰 등)",
    "Q8a18": "연수, 교육, 연구", "Q8a19": "기타"
}

q8b_code_mapping_2023 = {
    1: "식도락(음식/미식) 관광", 2: "쇼핑", 3: "자연경관 감상", 4: "고궁/역사 유적지 방문",
    5: "전통문화체험 (한복체험, 태권도, 한국음식 만들기 등)", 6: "박물관, 전시관 관람",
    7: "K-POP/한류스타 공연장 및 드라마/영화 촬영지 방문", 8: "연극, 뮤지컬, 발레 등 공연 관람",
    9: "지역 축제 참여", 10: "유흥(나이트라이프 활동 및 카지노 등)",
    11: "오락(놀이공원 및 테마파크 방문 등)", 12: "뷰티/미용 관광(마사지샵, 헤어샵, 네일케어 등 방문)",
    13: "치료/건강검진(피부과, 성형외과, 내외과 등 입원, 수술 등)", 14: "스포츠/레포츠 관람 (야구, 축구, 농구 등)",
    15: "스포츠/레포츠 참가 (스키, 수영, 골프, 캠핑 등)", 16: "국제회의(컨벤션, 국제회의, 전시 등)",
    17: "비즈니스 전문활동(업무회의, 산업시설 시찰 등)", 18: "연수, 교육, 연구", 19: "기타"
}

q8a_cols = list(q8a_mapping_2023.keys())
q8b_cols = ["Q8_1a1", "Q8_1a2", "Q8_1a3"]

# 2. 주요활동 추출 함수
def get_q8a_activities(row):
    return [q8a_mapping_2023[col] for col in q8a_cols if col in row and pd.notna(row[col])]

# 3. 만족활동 추출 함수
def get_q8b_satisfaction(row):
    return [q8b_code_mapping_2023.get(int(row[col]), f"코드 {row[col]}")
            for col in q8b_cols if col in row and pd.notna(row[col])]

# 4. 적용 (df_2023이 정의된 상태에서만)
df_2023["주요활동"] = df_2023.apply(get_q8a_activities, axis=1)
df_2023["만족활동"] = df_2023.apply(get_q8b_satisfaction, axis=1)

In [8]:
# 1. 활동명 매핑
q8a_mapping_2024 = {
    "Q8a01": "식도락(음식/미식) 관광", "Q8a02": "쇼핑", "Q8a03": "자연경관 감상", "Q8a04": "휴양/휴식(웰니스)",
    "Q8a05": "고궁/역사 유적지 방문", "Q8a06": "전통문화체험 (한복체험, 태권도, 한국 음식 만들기 등)",
    "Q8a07": "박물관, 전시관 관람", "Q8a08": "K-POP/한류스타 공연장 및 드라마/영화 촬영지 방문",
    "Q8a09": "연극, 뮤지컬, 발레 등 공연 관람", "Q8a10": "지역 축제 참여",
    "Q8a11": "유흥(나이트라이프 활동 및 카지노)", "Q8a12": "오락(놀이공원 및 테마파크 방문 등)",
    "Q8a13": "뷰티/미용 관광(마사지샵, 헤어샵, 네일케어 등 방문)", "Q8a14": "치료/건강검진(피부과, 성형외과, 내외과 입원, 수술 등)",
    "Q8a15": "스포츠/레포츠 관람 (야구, 축구, 농구 등)", "Q8a16": "스포츠/레포츠 참가 (스키, 수영, 골프, 캠핑 등)",
    "Q8a17": "국제회의(컨벤션, 국제회의, 전시 등)", "Q8a18": "비즈니스 전문활동(업무회의, 산업시설 시찰 등)",
    "Q8a19": "연수, 교육, 연구", "Q8a20": "기타"
}

q8b_code_mapping_2024 = {
    1: "식도락(음식/미식) 관광", 2: "쇼핑", 3: "자연경관 감상", 4: "휴양/휴식(웰니스)",
    5: "고궁/역사 유적지 방문", 6: "전통문화체험 (한복체험, 태권도, 한국 음식 만들기 등)",
    7: "박물관, 전시관 관람", 8: "K-POP/한류스타 공연장 및 드라마/영화 촬영지 방문",
    9: "연극, 뮤지컬, 발레 등 공연 관람", 10: "지역 축제 참여",
    11: "유흥(나이트라이프 활동 및 카지노)", 12: "오락(놀이공원 및 테마파크 방문 등)",
    13: "뷰티/미용 관광(마사지샵, 헤어샵, 네일케어 등 방문)", 14: "치료/건강검진(피부과, 성형외과, 내외과 입원, 수술 등)",
    15: "스포츠/레포츠 관람 (야구, 축구, 농구 등)", 16: "스포츠/레포츠 참가 (스키, 수영, 골프, 캠핑 등)",
    17: "국제회의(컨벤션, 국제회의, 전시 등)", 18: "비즈니스 전문활동(업무회의, 산업시설 시찰 등)",
    19: "연수, 교육, 연구", 20: "기타"
}

q8a_cols = list(q8a_mapping_2024.keys())
q8b_cols = ["Q8_1a1", "Q8_1a2", "Q8_1a3"]

# 2. 주요활동 추출 함수
def get_q8a_activities(row):
    return [q8a_mapping_2024[col] for col in q8a_cols if col in row and pd.notna(row[col])]

# 3. 만족활동 추출 함수
def get_q8b_satisfaction(row):
    return [q8b_code_mapping_2024.get(int(row[col]), f"코드 {row[col]}")
            for col in q8b_cols if col in row and pd.notna(row[col])]

# 4. 적용 (df_2024이 정의된 상태에서만 실행)
df_2024["주요활동"] = df_2024.apply(get_q8a_activities, axis=1)
df_2024["만족활동"] = df_2024.apply(get_q8b_satisfaction, axis=1)


In [9]:
# 클러스터1 국가 리스트
cluster1_countries = ["미국", "영국", "독일", "프랑스", "캐나다", "호주", "기타"]

# 공통 국가 필터링 함수
def filter_cluster1(df):
    return df[df["국가명"].isin(cluster1_countries)].copy()

# 연도별 필터링
df_cluster1_2022 = filter_cluster1(df_2022)
df_cluster1_2023 = filter_cluster1(df_2023)
df_cluster1_2024 = filter_cluster1(df_2024)

- 계절 컬럼 추가 (계절 분석 시 사용)

In [10]:
import pandas as pd

# 월 → 계절 매핑 함수
def month_to_season(month):
    if month in [3, 4, 5]:
        return 'Spring'
    elif month in [6, 7, 8]:
        return 'Summer'
    elif month in [9, 10, 11]:
        return 'Autumn'
    else:  # 12, 1, 2
        return 'Winter'

# 각 연도별 데이터프레임에 적용
for df in [df_cluster1_2024, df_cluster1_2023, df_cluster1_2022]:
    df['계절'] = df['D_MON'].apply(month_to_season)

# 📊 연관분석 실행 흐름 요약 (Weighted Apriori 기반)

## 1. 데이터 준비
- 2022~2024년 클러스터1 데이터 병합
- 필요한 컬럼만 추출: 방문자 ID, 국가명, 월→계절, 복수응답 활동, 가중치, 재방문 여부, FIT 여부 등

## 2. 트랜잭션 정의
- 한 명의 여행객을 하나의 트랜잭션으로 정의
- 트랜잭션당 아이템 리스트 구성: 활동, 계절, 국가 등 포함

## 3. 전처리
- 복수응답 문자열 → 리스트로 변환
- MultiLabelBinarizer 또는 수동 더미화로 One-Hot 인코딩
- 가중치 벡터(`weight`) 별도 보존

## 4. 가중치 반영 지지도 계산
- One-Hot 매트릭스 × 가중치로 weighted_df 생성
- 각 아이템의 weighted support 계산 (가중치 합 ÷ 전체 가중치)

## 5. 연관규칙 탐색 (Apriori 방식)
- 2개 아이템 조합별 joint support 계산
- confidence = 공동지지도 ÷ 단일지지도
- lift = confidence ÷ 대상 아이템 지지도
- 조건: support ≥ 0.05, confidence ≥ 0.6, lift > 1

## 6. 결과 정리
- 상위 lift 순으로 유의미한 규칙 10~20개 추출
- 재방문 여부, FIT 여부, 클러스터 그룹 등 기준으로 규칙 비교
- 해석 가능한 규칙을 전략 조합 후보로 저장

## 7. 다음 단계
- 파라미터 조정하며 규칙 확인
- 전략 조합별 재방문율 시뮬레이션으로 연결


# 1단계: 데이터 병합 및 주요 컬럼 추출

주요 컬럼 선정

> 참여활동 vs 만족활동: 무엇을 쓸까?

- 참여활동(Q8a01~Q8a20): 실질적으로 어떤 활동을 했는지 보여줌 → 실제 경험 기반 분석에 적합

- 만족활동(Q8_1a1~Q8_1a3): 참여한 것 중 가장 만족한 3가지를 고른 것 → 고객의 긍정적 감정·충성도와 더 밀접

-------

**필수 항목 (트랜잭션 구성 요소)**

- Q8_1a1 ~ Q8_1a3(만족활동 컬럼 정리 되어있음): 만족한 활동 리스트 (중심 아이템)

- D_MON → 계절 태그 생성 (Spring, Summer, …)

- D_NAT: 국가명

- weight: 가중치

**보조적 그룹 비교용 메타데이터**
(트랜잭션에 넣지 않되, 분석 시 그룹 나누기에 사용)

- is_returning (재방문 여부)

- is_FIT (자유여행객 여부)

- D_AGE, D_SEX: 인구통계

- 총액1인TOT2 또는 C총액1인TOT2: 1인 소비 총액
-------

**2022년은 가중치 정보가 없어 본 분석에는 포함하지 않았으며, 트렌드 비교용으로만 활용하였습니다.**

| 표준 컬럼명          | 설명               | 원본 컬럼 (2023/2024)  |
| --------------- | ---------------- | ------------------ |
| `visitor_id`    | 방문객 고유 ID        | `pnid`             |
| `만족활동`          | 리스트형 활동 조합       | `만족활동` (이미 리스트 형태) |
| `weight`        | 응답자 가중치          | `weight`           |
| `계절`            | 계절 (Spring 등)    | `계절`               |
| `D_NAT`         | 국가명              | `D_NAT`            |
| `is_FIT`        | 개별여행 여부 (1 = 개별) | `TYP`              |
| `revisit_score` | 재방문의사 점수 (1\~5점) | `Q13`              |


In [28]:
# 2023 데이터 정제
df_2023 = df_cluster1_2023.copy()
df_2023['visitor_id'] = df_2023['pnid']
df_2023['is_FIT'] = (df_2023['TYP'] == 1).astype(int)
df_2023['revisit_score'] = df_2023['Q13']
df_2023 = df_2023[['visitor_id', '만족활동', 'weight', '계절', '국가명', 'is_FIT', 'revisit_score']]

# 2024 데이터 정제
df_2024 = df_cluster1_2024.copy()
df_2024['visitor_id'] = df_2024['pnid']
df_2024['is_FIT'] = (df_2024['TYP'] == 1).astype(int)
df_2024['revisit_score'] = df_2024['Q13']
df_2024 = df_2024[['visitor_id', '만족활동', 'weight', '계절', '국가명', 'is_FIT', 'revisit_score']]

# 병합
df_all = pd.concat([df_2023, df_2024], ignore_index=True)

# 2단계: item_list 생성 + One-hot 인코딩 + 가중치 적용

## 2-1. item_list 생성

만족활동 + 계절 + 국가명을 하나의 리스트로 묶어줍니다.

In [30]:
# 리스트 결합: 만족활동 + 계절 + 국가명
# 각 행 단위로 리스트 결합
df_all['item_list'] = df_all.apply(
    lambda row: row['만족활동'] + [row['계절'], row['국가명']],
    axis=1
)

## 2-2. One-hot 인코딩 (MultiLabelBinarizer 사용)

연관분석(Apriori)은 트랜잭션별로 어떤 "아이템"이 있었는지 **0/1로 표현된 테이블(행렬)**을 요구

즉, 각 방문객이 어떤 활동/계절/국가를 가졌는지 True/False 또는 1/0으로 변환해야 분석이 가능

In [31]:
# 모든 아이템을 str로 변환
df_all['item_list'] = df_all['item_list'].apply(lambda items: [str(i) for i in items])

In [32]:
from sklearn.preprocessing import MultiLabelBinarizer

mlb = MultiLabelBinarizer()
df_hot = pd.DataFrame(
    mlb.fit_transform(df_all['item_list']),
    columns=mlb.classes_,
    index=df_all.index
)

## 2-3. 가중치 반영 지지도 계산

단순 빈도(count)가 아닌, **응답자 가중치(weight)**를 반영하여 정확한 지지도(support)를 계산하기 위함

외래관광객 실태조사는 표본조사이기 때문에,
한 명의 응답자가 전체 외래관광객을 대표할 수 있도록 weight을 부여

In [33]:
# 가중치 벡터
weight = df_all['weight']
total_weight = weight.sum()

# 가중치 곱한 매트릭스
weighted_df = df_hot.multiply(weight, axis=0)

# 연관분석 3단계: 가중치 기반 연관규칙 계산

## 3-1. 단일 아이템별 weighted support 계산

- 목적: 각 아이템이 전체 응답자 중에서 얼마나 많이 등장했는지를 **가중치까지 반영해서 비율**로 계산

      weighted support(A) = A를 선택한 응답자들의 weight 합 / 전체 weight 합

w_sup[item]: 특정 아이템이 전체 응답자 가중치에서 차지하는 비율

In [34]:
# 전체 가중치 총합
total_weight = df_all['weight'].sum()

# 각 아이템의 weighted support 계산
w_sup = {
    item: weighted_df[item].sum() / total_weight
    for item in df_hot.columns
}

## 3-2. 아이템쌍별 joint support 계산

- 목적 : 두 아이템 A, B가 같은 응답자(트랜잭션) 안에서 **함께 등장한 비율** 계산

      " A와 B를 동시에 선택한 응답자들의 비율은 전체 중 몇 퍼센트인가? "

      joint support(A, B) = A와 B를 모두 선택한 응답자들의 가중치 합 ÷ 전체 가중치 합

여기서도 **단순 건수(count)**가 아니라, **가중치(weight)**를 고려한 비율을 사용

------

[코드 리뷰]

- df_hot[i] & df_hot[j]: 두 아이템을 모두 선택한 응답자 (0 또는 1)

- multiply(...): 해당 응답자의 가중치를 곱함

- `w_sup_pair`는 **아이템쌍별 joint support 저장** 딕셔너리

In [35]:
w_sup_pair = {}
for i in df_hot.columns:
    for j in df_hot.columns:
        if i == j:
            continue # 자기 자신 쌍은 제외
        pair = (i, j)
        # i와 j를 모두 선택한 응답자: (0/1) AND 연산
        joint = (df_hot[i] & df_hot[j]).multiply(df_all['weight'], axis=0)
        # joint support 계산
        w_sup_pair[pair] = joint.sum() / total_weight

## 3-3. Confidence 계산

- 목적 : 연관규칙 A → B에서, “**A를 선택한 사람들 중에 B도 함께 선택한 비율**”을 구합니다.

- 의미: i → j 규칙의 신뢰도 (i가 발생했을 때 j도 함께 발생할 확률)

----------

[코드 구성]

| 요소                 | 의미                                 |
| ------------------ | ---------------------------------- |
| `pair`             | (A, B) 형식의 아이템쌍                    |
| `w_sup_pair[pair]` | A와 B가 동시에 등장한 weighted support     |
| `w_sup[pair[0]]`   | A 단독의 weighted support             |
| 결과값                | A를 선택한 사람 중 B도 선택한 비율 → confidence |

해석 방법

- 값이 1에 가까울수록: A가 나타났을 때 거의 항상 B도 나타난다

- 값이 0.5면: 절반 정도만 함께 나타난다

In [36]:
w_conf = {
    pair: w_sup_pair[pair] / w_sup[pair[0]]
    for pair in w_sup_pair
}


## 3-4. Lift 계산

- 목적 : 연관규칙 A → B가 **우연이 아니라 실제로 강한 연관성을 가지는가?**를 평가

   Lift는 통계적으로 “A와 B가 독립일 때 기대되는 확률 대비 얼마나 더 자주 같이 나타나는가”를 수치화한 지표입니다.

   단순히 j가 나타나는 확률에 비해, i가 있을 때 j가 얼마나 더 자주 발생하는지 비율

      수식: Lift(A→B)= Confidence(A→B)/Support(B)

      - confidence(A→B)는 A 했을 때 B도 하는 비율

      - support(B)는 B 자체가 전체에서 얼마나 자주 나타나는지


| Lift 값 | 해석                                |
| ------ | --------------------------------- |
| > 1    | **양의 연관** (A가 있을 때 B가 더 자주 나타남) ✅ |
| = 1    | **독립** (A와 B는 무관함) ⚠️             |
| < 1    | **음의 연관** (A가 있을 때 B가 덜 나타남) ❌    |

-----------

[코드 구성]

| 요소               | 의미                |
| ---------------- | ----------------- |
| `pair`           | (A, B) 형식의 아이템쌍   |
| `w_conf[pair]`   | confidence(A→B)   |
| `w_sup[pair[1]]` | support(B)        |
| 결과값              | A와 B의 연관 강도 판단 수치 |



In [37]:
w_lift = {
    pair: w_conf[pair] / w_sup[pair[1]]
    for pair in w_conf
}

## 3-5. 유의미한 규칙 필터링

| 지표             | 권장 기준값         | 설명 및 출처                    |
| -------------- | -------------- | -------------------------- |
| **Support**    | ≥ 0.05 (5%)    | 트랜잭션 중 5% 이상 등장하는 항목만 포함   |
| **Confidence** | ≥ 0.5 또는 ≥ 0.6 | 50〜60% 이상이면 실무 평가 기준으로 적합  |
| **Lift**       | > 1.0          | 독립 관계 대비 강화된 연관성 판단 기준     |


**왜 이 기준을 사용하나요?**

- Support ≥ 5%

  너무 낮은 기준은 노이즈나 희귀 패턴까지 포함할 위험 있음

- Confidence ≥ 50–60%

  A를 한 사람 중 절반 이상이 B를 하는 수준이면 의미 있는 연관으로 간주됨

- Lift > 1.0

  단순 우연이 아닌 실질적 상관관계가 존재한다고 보는 기준임

In [38]:
min_support = 0.05
min_confidence = 0.6

# 조건을 만족하는 규칙만 필터링
selected_rules = [
    (pair, w_sup_pair[pair], w_conf[pair], w_lift[pair])
    for pair in w_conf
    if w_sup_pair[pair] >= min_support
    and w_conf[pair] >= min_confidence
    and w_lift[pair] > 1
]

- 결과 정렬 및 출력 (lift 기준 상위 10개)

In [39]:
# lift 높은 순으로 정렬
sorted_rules = sorted(selected_rules, key=lambda x: x[3], reverse=True)

# 상위 10개 규칙 출력
for rule in sorted_rules[:10]:
    A, B = rule[0]
    support, conf, lift = rule[1], rule[2], rule[3]
    print(f"{A} → {B} | support: {support:.3f}, confidence: {conf:.3f}, lift: {lift:.2f}")


캐나다 → 쇼핑 | support: 0.099, confidence: 0.652, lift: 1.35


### 결과

| 항목                    | 의미                                         |
| --------------------- | ------------------------------------------ |
| **캐나다**               | 캐나다 국적의 응답자들이                              |
| **→ 쇼핑**              | **쇼핑 활동을 할 가능성이 높음**                       |
| **support: 0.099**    | 전체 응답자의 약 \*\*9.9%\*\*가 캐나다 국적 + 쇼핑 선택     |
| **confidence: 0.652** | 캐나다 응답자 중 **약 65.2%가 쇼핑 선택**               |
| **lift: 1.35**        | **기대보다 35% 더 많이 쇼핑을 선택**함 → 마케팅 타깃 가치 있음 ✅ |

## 기준 완화 적용 코드 (탐색 확장 목적)

support ≥ 0.03, confidence ≥ 0.5, lift > 1

In [40]:
# 완화된 기준 설정
min_support = 0.03
min_confidence = 0.5

# 필터링: support ≥ 0.03, confidence ≥ 0.5, lift > 1.0
selected_rules = [
    (pair, w_sup_pair[pair], w_conf[pair], w_lift[pair])
    for pair in w_conf
    if w_sup_pair[pair] >= min_support
    and w_conf[pair] >= min_confidence
    and w_lift[pair] > 1.0
]

# 정렬 및 출력
sorted_rules = sorted(selected_rules, key=lambda x: x[3], reverse=True)

for rule in sorted_rules[:10]:
    A, B = rule[0]
    support, conf, lift = rule[1], rule[2], rule[3]
    print(f"{A} → {B} | support: {support:.3f}, confidence: {conf:.3f}, lift: {lift:.2f}")

전통문화체험 (한복체험, 태권도, 한국 음식 만들기 등) → 기타 | support: 0.055, confidence: 0.514, lift: 1.40
캐나다 → 쇼핑 | support: 0.099, confidence: 0.652, lift: 1.35
프랑스 → 쇼핑 | support: 0.054, confidence: 0.586, lift: 1.21
호주 → 쇼핑 | support: 0.067, confidence: 0.581, lift: 1.20
박물관, 전시관 관람 → 고궁/역사 유적지 방문 | support: 0.086, confidence: 0.535, lift: 1.19
독일 → 쇼핑 | support: 0.052, confidence: 0.562, lift: 1.17
전통문화체험 (한복체험, 태권도, 한국음식 만들기 등) → 고궁/역사 유적지 방문 | support: 0.034, confidence: 0.521, lift: 1.16
기타 → 고궁/역사 유적지 방문 | support: 0.189, confidence: 0.517, lift: 1.15
전통문화체험 (한복체험, 태권도, 한국 음식 만들기 등) → 고궁/역사 유적지 방문 | support: 0.055, confidence: 0.513, lift: 1.14
미국 → 쇼핑 | support: 0.067, confidence: 0.543, lift: 1.13


### 결과

| 연관규칙                   | 해석                                   | 전략 활용 가능성                    |
| ---------------------- | ------------------------------------ | ---------------------------- |
| **전통문화체험 → 기타**        | 전통문화 콘텐츠 경험자들이 기타 활동(예: 이색체험)도 자주 수행 | 전통 + 이색체험 묶음 상품 제안 가능        |
| **캐나다 → 쇼핑**           | 캐나다 관광객은 쇼핑 선호 경향이 뚜렷                | 쇼핑 중심 콘텐츠 큐레이션 추천            |
| **프랑스/호주/독일/미국 → 쇼핑**  | 쇼핑은 장거리 고소비국의 공통 관심사                 | 쇼핑 콘텐츠는 **클러스터 전체 전략**으로 유효함 |
| **박물관 관람 → 고궁/역사 유적지** | 역사적 관심이 이어짐                          | 박물관 + 고궁 연계 패스 제안 전략         |
| **전통문화체험 → 고궁/역사 유적지** | 전통 콘텐츠에 연속 관심                        | 콘텐츠 큐레이션, 마케팅 카피 활용 전략       |

<br>

① [전통문화체험 → 고궁/역사 유적지 방문]

  - lift = 1.16 / confidence = 0.521

  → 고궁 야간개장, 전통행사 이벤트 등과 연계한 상품 추천 가능

  → K-문화 강조 캠페인과도 잘 맞음

<br>

② [국가 → 쇼핑] 연관

- 캐나다, 프랑스, 호주, 독일, 미국 모두 lift > 1.1

  → "쇼핑 콘텐츠 큐레이션은 장거리 국가 공통 타깃"이라는 근거로 사용 가능

  → 국가별 마케팅 메시지 차별화 없이도 적용 가능


# 세그먼트별 비교 (재방문, FIT)

→ 동일 규칙이 어떤 그룹에서 더 강한지 확인
<br>

| 구분                            | 의미                        | 우리가 알 수 있는 것              | 분석 목적 적합성               |
| ----------------------------- | ------------------------- | ------------------------- | ----------------------- |
| **is\_returning (실제 재방문 여부)** | 과거에 한 번 이상 한국에 다시 온 사람    | “재방문자들이 뭘 했는지”            | ❌ 제한적. **미래 전략 연결 어려움** |
| **revisit\_score (1\~5점)**    | 이번 방한 후 향후 다시 방문할 의향이 있는지 | **“어떤 요소가 재방문 의사를 높이는지”** | ✅ 전략 타깃 예측에 적합          |


- 1. 세그먼트 분할

| 그룹명            | 조건                                  |
| -------------- | ----------------------------------- |
| `재방문 의향 + FIT` | `revisit_score ≥ 4` & `is_FIT == 1` |
| `재방문 의향 + 패키지` | `revisit_score ≥ 4` & `is_FIT == 0` |
| `비재방문 + FIT`   | `revisit_score < 4` & `is_FIT == 1` |
| `비재방문 + 패키지`   | `revisit_score < 4` & `is_FIT == 0` |

In [41]:
df_positive_fit  = df_all[(df_all['revisit_score'] >= 4) & (df_all['is_FIT'] == 1)].copy()
df_positive_pkg  = df_all[(df_all['revisit_score'] >= 4) & (df_all['is_FIT'] == 0)].copy()
df_negative_fit  = df_all[(df_all['revisit_score'] < 4) & (df_all['is_FIT'] == 1)].copy()
df_negative_pkg  = df_all[(df_all['revisit_score'] < 4) & (df_all['is_FIT'] == 0)].copy()

- 2. 연관분석 함수 재사용

In [54]:
from sklearn.preprocessing import MultiLabelBinarizer

def run_association_analysis(df, min_support=0.05, min_confidence=0.5):
    """
    가중치를 반영한 연관분석(Apriori) 함수
    - 입력: df (item_list, weight 포함된 DataFrame)
    - 출력: (item A → B, support, confidence, lift) 리스트 (상위 10개)

    Parameters:
        df: pandas DataFrame
            - 필수 컬럼: item_list (리스트), weight (가중치)
        min_support: float
            - 최소 지지도 기준
        min_confidence: float
            - 최소 신뢰도 기준
    """

    # 1. 리스트 항목을 One-Hot Encoding (0/1 매트릭스로 변환)
    mlb = MultiLabelBinarizer()
    df_items = mlb.fit_transform(df['item_list'])  # 예: ['쇼핑', '전통'] → [1, 0, 1, ...]
    df_hot = pd.DataFrame(df_items, columns=mlb.classes_, index=df.index)

    # 2. 가중치 적용
    # 각 항목에 weight 곱해서 가중 합을 구함
    weighted_df = df_hot.multiply(df['weight'], axis=0)
    total_weight = df['weight'].sum()

    # 3. 각 단일 아이템별 Weighted Support 계산
    w_sup = {
        item: weighted_df[item].sum() / total_weight
        for item in df_hot.columns
    }

    # 4. 아이템 쌍별(2-item) 조합의 Joint Weighted Support 계산
    w_sup_pair = {}
    for i in df_hot.columns:
        for j in df_hot.columns:
            if i == j:
                continue
            # 두 항목이 동시에 선택된 경우만 추출 후 weight 곱함
            joint = (df_hot[i] & df_hot[j]).multiply(df['weight'], axis=0)
            w_sup_pair[(i, j)] = joint.sum() / total_weight

    # 5. Weighted Confidence 계산
    w_conf = {
        pair: w_sup_pair[pair] / w_sup[pair[0]]
        for pair in w_sup_pair
    }

    # 6. Weighted Lift 계산
    w_lift = {
        pair: w_conf[pair] / w_sup[pair[1]]
        for pair in w_conf
    }

    # 7. 규칙 필터링 (조건 만족하는 연관 규칙만 추출)
    rules = [
        (pair, w_sup_pair[pair], w_conf[pair], w_lift[pair])
        for pair in w_conf
        if w_sup_pair[pair] >= min_support      # 지지도 기준
        and w_conf[pair] >= min_confidence     # 신뢰도 기준
        and w_lift[pair] > 1                   # 양의 연관만 추출
    ]

    # 8. Lift 기준으로 내림차순 정렬 후 상위 10개 반환
    return sorted(rules, key=lambda x: x[3], reverse=True)[:10]


- 3. 4개 그룹 실행

In [55]:
segments = {
    "재방문 의향 + FIT": df_positive_fit,
    "재방문 의향 + 패키지": df_positive_pkg,
    "비재방문 + FIT": df_negative_fit,
    "비재방문 + 패키지": df_negative_pkg
}

results = {}
for name, df_segment in segments.items():
    print(f"\n 세그먼트: {name}")
    rules = run_association_analysis(df_segment)
    results[name] = rules
    for r in rules:
        a, b = r[0]
        print(f"{a} → {b} | support: {r[1]:.3f}, confidence: {r[2]:.3f}, lift: {r[3]:.2f}")


 세그먼트: 재방문 의향 + FIT
전통문화체험 (한복체험, 태권도, 한국 음식 만들기 등) → 기타 | support: 0.059, confidence: 0.583, lift: 1.52
캐나다 → 쇼핑 | support: 0.107, confidence: 0.686, lift: 1.39
박물관, 전시관 관람 → 기타 | support: 0.081, confidence: 0.519, lift: 1.35
호주 → 쇼핑 | support: 0.058, confidence: 0.616, lift: 1.25
박물관, 전시관 관람 → 고궁/역사 유적지 방문 | support: 0.081, confidence: 0.524, lift: 1.20
전통문화체험 (한복체험, 태권도, 한국 음식 만들기 등) → 고궁/역사 유적지 방문 | support: 0.052, confidence: 0.516, lift: 1.18
기타 → 고궁/역사 유적지 방문 | support: 0.197, confidence: 0.514, lift: 1.17
독일 → 쇼핑 | support: 0.058, confidence: 0.578, lift: 1.17
미국 → 식도락(음식/미식) 관광 | support: 0.060, confidence: 0.506, lift: 1.16
미국 → 쇼핑 | support: 0.067, confidence: 0.567, lift: 1.15

 세그먼트: 재방문 의향 + 패키지
프랑스 → 쇼핑 | support: 0.132, confidence: 0.675, lift: 1.29
기타 → 고궁/역사 유적지 방문 | support: 0.056, confidence: 0.699, lift: 1.27
캐나다 → 식도락(음식/미식) 관광 | support: 0.064, confidence: 0.522, lift: 1.15
Summer → 자연경관 감상 | support: 0.083, confidence: 0.502, lift: 1.12
Spring → 식도락(음식/미식) 관광 |

## 재방문 의향이 4-5점인 사람들만 전략 제시(기타활동 포함시 전략에서 제외)

대상: revisit_score ≥ 4 (재방문 의향 있음)

세분: FIT(개별여행객), 패키지(단체여행객)

분석 방식: 연관분석 (support ≥ 0.03, confidence ≥ 0.5, lift > 1)

가중치 반영


**세그먼트 1: 재방문 의향 + FIT(개별여행)**

| 연관규칙                       | support | confidence | lift     | 전략 해석                  |
| -------------------------- | ------- | ---------- | -------- | ---------------------- |
| 캐나다 → 쇼핑                   | 0.107   | 0.686      | **1.39** | 캐나다 고객 대상 쇼핑 연계 패스     |
| 호주 → 쇼핑                    | 0.058   | 0.616      | **1.25** | 호주 FIT 고객 대상 쇼핑 콘텐츠 기획 |
| 박물관, 전시관 관람 → 고궁/역사 유적지 방문 | 0.081   | 0.524      | **1.20** | 박물관 관람 후 역사 콘텐츠 유도     |
| 전통문화체험 → 고궁/역사 유적지 방문      | 0.052   | 0.516      | **1.18** | 체험형 콘텐츠와 역사 유산의 연계 기획  |
| 독일 → 쇼핑                    | 0.058   | 0.578      | **1.17** | 독일 고객 대상 면세/쇼핑 테마 강화   |
| 프랑스 → 식도락 관광               | 0.044   | 0.512      | **1.17** | 프랑스 고객 대상 미식 콘텐츠 큐레이션  |
| 미국 → 식도락 관광                | 0.060   | 0.506      | **1.16** | 미국 고객 대상 먹거리 투어 구성     |



“전통 → 역사” 흐름은 행동 흐름 중심 전략에 적합

“국가 → 쇼핑/식도락”은 국가별 테마 설계에 활용 가능

<br>

**세그먼트 2: 재방문 의향 + non-FIT**

| 연관규칙                  | support | confidence | lift     | 전략 해석                 |
| --------------------- | ------- | ---------- | -------- | --------------------- |
| 스포츠/레포츠 참가 → Winter   | 0.031   | 0.633      | **2.85** | 겨울 시즌 스포츠 특화 패키지 구성   |
| 프랑스 → 쇼핑              | 0.132   | 0.675      | **1.29** | 프랑스 고객 대상 쇼핑 중심 일정 설계 |
| 캐나다 → 식도락 관광          | 0.064   | 0.522      | **1.15** | 캐나다 대상 미식 중심 체험 콘텐츠   |
| Summer → 자연경관 감상      | 0.083   | 0.502      | **1.12** | 여름 휴양형 자연 중심 콘텐츠 구성   |
| Spring → 식도락 관광       | 0.134   | 0.503      | **1.10** | 봄 시즌 미식 투어 집중 기획      |
| 캐나다 → 고궁/역사 유적지 방문    | 0.073   | 0.589      | **1.07** | 문화 유산 콘텐츠에 대한 관심 반영   |
| Spring → 고궁/역사 유적지 방문 | 0.156   | 0.583      | **1.06** | 봄 시즌 역사 관광 연계 콘텐츠 강화  |
| Autumn → 쇼핑           | 0.191   | 0.554      | **1.06** | 가을철 쇼핑 마케팅 타이밍 연계     |



“계절+콘텐츠 조합”이 강하게 나타남 → 시즌 특화형 패키지 전략 설계 유리

프랑스/캐나다 고객의 음식·쇼핑 선호 반영 필요


## support 지수에 대한 고민!

support = 0.03 이란?
→ 전체 가중치 기준 3%의 응답자가 해당 연관규칙에 해당된다는 의미입니다.

예를 들어, 응답자 가중치 총합이 10,000명이라면 → 약 300명 수준.

<br>

**믿을만한 기준인가?**

| 관점          | 판단                                                                            |
| ----------- | ----------------------------------------------------------------------------- |
| **이론적 해석**  | 0.03 이상이면 ‘희귀 규칙’은 아니며, **패턴 탐지 가능 수준**입니다.                                   |
| **실무적 판단**  | 전략 기획에 활용하기 위해선 **lift나 confidence가 충분히 높아야** 신뢰할 수 있습니다.                     |
| **통계적 신뢰도** | support가 너무 낮으면 → 통계적 변동성(노이즈)에 **민감**합니다.                                    |
| **실제 사용 예** | 마케팅, 관광, 유통 등에서는 0.02\~0.05 사이도 **자주 사용**됩니다. 단, **다른 조건(신뢰도, lift)을 강화**합니다. |


**결론**

support 0.03은 사용 가능하지만, confidence ≥ 0.5, lift > 1.2 이상일 때만 전략적으로 의미 있다고 간주하는 것이 안전

support 크기에 따라 lift 기준을 조절하기로!!

# 전략 설계: 세그먼트별 연관분석 기반

---

## 1. FIT(개별여행객) 대상 “문화·역사 연계” 전략

**핵심 패턴**  
- 박물관·전시관 관람 → 고궁/역사유적지 방문  
- 전통문화체험 → 고궁/역사유적지 방문  

**전략 기획 예시**  
- **서울 문화유산 1일 투어**  
  - 오전: 국립중앙박물관 → 오후: 경복궁 한복체험 → 북촌 산책  
- **체험형 역사여행 패스**  
  - 전통 한식 만들기 → 태권도 체험 → 경복궁 야간개장  

---

## 2. non-FIT(패키지여행객) 대상 “계절별 테마” 전략

| 계절      | 연관 패턴                        | 전략 아이디어                                |
|-----------|---------------------------------|--------------------------------------------|
| **Winter** | 스포츠/레포츠 참가 → Winter       | “겨울 레포츠 패키지” (스키 + 온천투어)               |
| **Spring** | Spring → 고궁/역사유적지 방문     | “봄꽃 역사기행” (창덕궁 벚꽃길 + 전통공예 체험)       |
| **Summer** | Summer → 자연경관 감상           | “여름 자연힐링 투어” (섬·계곡·숲속 글램핑)            |
| **Autumn** | Autumn → 쇼핑                   | “가을 쇼핑 페스티벌” (전통시장 + 아울렛 투어)         |

---

## 3. 실무 적용 팁

1. **프로모션 타이밍**  
   - 시즌 시작 전 얼리버드 프로모션(1~2개월 전)  
   - 시즌 중반엔 야간개장·축제 옵션 A/B 테스트  

2. **크로스셀 & 업셀**  
   - FIT 투어에 “역사+체험” 옵션 추가  
   - 패키지에 “숙소 업그레이드”·“현지 레스토랑 체험” 묶음  

3. **A/B 테스트**  
   - 그룹별 광고 메시지(문화·역사 vs 계절 테마) 집행  
   - 전환율·예약 전환율 비교 분석  

---


# 검증

## 1. 데이터 준비

- revisit_score ≥ 4 & is_FIT 기준으로 두 DataFrame 생성

- item_list → df_hot (0/1 매트릭스)

In [56]:
from sklearn.preprocessing import MultiLabelBinarizer
import numpy as np
import pandas as pd

# 1-1. 그룹 필터링
df_fit     = df_all[(df_all['revisit_score'] >= 4) & (df_all['is_FIT'] == 1)].reset_index(drop=True)
df_nonfit  = df_all[(df_all['revisit_score'] >= 4) & (df_all['is_FIT'] == 0)].reset_index(drop=True)

# 1-2. One-hot 인코딩 함수
def make_hot(df):
    mlb = MultiLabelBinarizer()
    hot = pd.DataFrame(
        mlb.fit_transform(df['item_list']),
        columns=mlb.classes_,
        index=df.index
    )
    return hot

# 1-3. 그룹별 hot/weights 준비
df_hot_fit    = make_hot(df_fit)
weights_fit   = df_fit['weight']
df_hot_nonfit = make_hot(df_nonfit)
weights_nonfit= df_nonfit['weight']

## 2. 연관 지표 함수 정의

- rule_metrics(): support, confidence, lift 반환

In [57]:
def rule_metrics(hot, w, ant, con):
    """주어진 antecedent→consequent에 대한 support, confidence, lift 계산."""
    total_w = w.sum()
    sup_ac  = (hot[ant] & hot[con]).mul(w).sum() / total_w
    sup_a   = hot[ant].mul(w).sum() / total_w
    sup_c   = hot[con].mul(w).sum() / total_w
    conf    = sup_ac / sup_a if sup_a>0 else np.nan
    lift    = conf / sup_c if sup_c>0 else np.nan
    return sup_ac, conf, lift

## 3. Bootstrap 95% CI 함수

- bootstrap_ci(): 1,000회 부트스트랩 샘플 → 2.5%–97.5% 분위수 계산

In [58]:
def bootstrap_ci(hot, w, ant, con, n_iter=1000):
    """Bootstrap으로 support, confidence, lift 95% CI 계산."""
    np.random.seed(0)
    metrics = []
    n = len(hot)
    for _ in range(n_iter):
        idx = np.random.choice(n, n, replace=True)
        sup, conf, lift = rule_metrics(hot.iloc[idx], w.iloc[idx], ant, con)
        metrics.append((sup, conf, lift))
    dfm = pd.DataFrame(metrics, columns=['support','confidence','lift'])
    return dfm.quantile([0.025, 0.975])

## 4. Permutation Test 함수

- permutation_pvalue(): 귀무가설 하 permuted lift 분포 생성 → p-value 산출

In [61]:
def permutation_pvalue(hot, w, ant, con, n_iter=1000):
    """
    Permutation test:
    - 귀무가설: ant와 con이 독립.
    - p-value: 관측 lift가 permuted lift보다 큰 비율.
    """
    obs_sup, obs_conf, obs_lift = rule_metrics(hot, w, ant, con)
    count = 0
    n = len(hot)
    for _ in range(n_iter):
        perm_con = hot[con].sample(frac=1, replace=False).reset_index(drop=True)
        sup, conf, lift = rule_metrics(
            hot[[ant]].reset_index(drop=True).join(perm_con),
            w.reset_index(drop=True),
            ant, con
        )
        if lift >= obs_lift:
            count += 1
    return count / n_iter

## 5. 검증할 전략 패턴 리스트

- FIT 그룹 2개, non-FIT 그룹 4개 패턴

In [65]:
fit_rules = [
    ("박물관, 전시관 관람", "고궁/역사 유적지 방문"),
    ("전통문화체험 (한복체험, 태권도, 한국음식 만들기 등)", "고궁/역사 유적지 방문"),
]
nonfit_rules = [
    ("스포츠/레포츠 참가 (스키, 수영, 골프, 캠핑 등)", "Winter"),
    ("Spring", "고궁/역사 유적지 방문"),
    ("Summer", "자연경관 감상"),
    ("Autumn", "쇼핑"),
]

## 6. Bootstrap CI & Permutation p-value 계산 및 출력

- 각 규칙별 CI와 p-value 확인 →

- CI 하한 > 1이면 “양의 연관이 통계적으로 안정적”

- p-value < 0.05이면 “우연일 확률 낮음”

In [64]:
# df_hot_fit 에 어떤 아이템이 있는지 리스트로 출력해 봅니다
print(df_hot_fit.columns.tolist())

['Autumn', 'K-POP/한류스타 공연장 및 드라마/영화 촬영지 방문', 'Spring', 'Summer', 'Winter', '고궁/역사 유적지 방문', '국제회의(컨벤션, 국제회의, 전시 등)', '기타', '독일', '미국', '박물관, 전시관 관람', '뷰티/미용 관광(마사지샵, 헤어샵, 네일케어 등 방문)', '비즈니스 전문활동(업무회의, 산업시설 시찰 등)', '쇼핑', '스포츠/레포츠 관람 (야구, 축구, 농구 등)', '스포츠/레포츠 참가 (스키, 수영, 골프, 캠핑 등)', '식도락(음식/미식) 관광', '연극, 뮤지컬, 발레 등 공연 관람', '연수, 교육, 연구', '영국', '오락(놀이공원 및 테마파크 방문 등)', '유흥(나이트라이프 활동 및 카지노 등)', '유흥(나이트라이프 활동 및 카지노)', '자연경관 감상', '전통문화체험 (한복체험, 태권도, 한국 음식 만들기 등)', '전통문화체험 (한복체험, 태권도, 한국음식 만들기 등)', '지역 축제 참여', '치료/건강검진(피부과, 성형외과, 내외과 등 입원, 수술 등)', '치료/건강검진(피부과, 성형외과, 내외과 입원, 수술 등)', '캐나다', '프랑스', '호주', '휴양/휴식(웰니스)']


In [66]:
# 6-1. FIT 그룹 검증
print("▶ FIT 그룹 전략 검증")
for ant, con in fit_rules:
    ci = bootstrap_ci(df_hot_fit, weights_fit, ant, con)
    p = permutation_pvalue(df_hot_fit, weights_fit, ant, con)
    print(f"\n{ant} → {con}")
    print(ci)
    print(f"Permutation p-value: {p:.3f}")

# 6-2. non-FIT 그룹 검증
print("\n▶ non-FIT 그룹 전략 검증")
for ant, con in nonfit_rules:
    ci = bootstrap_ci(df_hot_nonfit, weights_nonfit, ant, con)
    p = permutation_pvalue(df_hot_nonfit, weights_nonfit, ant, con)
    print(f"\n{ant} → {con}")
    print(ci)
    print(f"Permutation p-value: {p:.3f}")

▶ FIT 그룹 전략 검증

박물관, 전시관 관람 → 고궁/역사 유적지 방문
        support  confidence      lift
0.025  0.071740    0.481166  1.108581
0.975  0.091462    0.566273  1.287670
Permutation p-value: 0.000

전통문화체험 (한복체험, 태권도, 한국음식 만들기 등) → 고궁/역사 유적지 방문
        support  confidence      lift
0.025  0.023513    0.416829  0.954692
0.975  0.034208    0.547604  1.242213
Permutation p-value: 0.086

▶ non-FIT 그룹 전략 검증

스포츠/레포츠 참가 (스키, 수영, 골프, 캠핑 등) → Winter
        support  confidence      lift
0.025  0.019718    0.487949  2.196760
0.975  0.044087    0.770621  3.579682
Permutation p-value: 0.000

Spring → 고궁/역사 유적지 방문
        support  confidence      lift
0.025  0.129155    0.513597  0.952510
0.975  0.184369    0.652012  1.180285
Permutation p-value: 0.183

Summer → 자연경관 감상
        support  confidence      lift
0.025  0.064828    0.427027  0.954466
0.975  0.104132    0.577537  1.291499
Permutation p-value: 0.083

Autumn → 쇼핑
        support  confidence      lift
0.025  0.157746    0.481208  0.943618
0.975  0.226934

## 결과

| 연관규칙                                                   | 그룹       | 95% CI (lift)    | p-value | 결정       |
|-------------------------------------------------------------|-----------|------------------|---------|-----------|
| 박물관·전시관 관람 → 고궁/역사 유적지 방문                   | FIT       | [1.109, 1.288]   | 0.000   | 강력 추천 |
| 전통문화체험 (한복체험, 태권도, 한국음식 만들기 등) → 고궁/역사 유적지 방문 | FIT       | [0.955, 1.242]   | 0.086   | 보조 전략 |
| 스포츠/레포츠 참가 (스키, 수영, 골프, 캠핑 등) → Winter      | non-FIT   | [2.197, 3.580]   | 0.000   | 강력 추천 |
| Spring → 고궁/역사 유적지 방문                              | non-FIT   | [0.953, 1.180]   | 0.183   | 제외       |
| Summer → 자연경관 감상                                      | non-FIT   | [0.954, 1.291]   | 0.083   | 보조 전략 |
| Autumn → 쇼핑                                              | non-FIT   | [0.944, 1.167]   | 0.147   | 제외       |

▶ 결론

- 통계적으로 강력 검증된 전략

  - 박물관 관람 → 고궁/역사 (FIT 그룹)

  - 스포츠/레포츠 참가 → Winter (non-FIT 그룹)

- 보조/재검토 전략

  - 전통문화체험 → 고궁 (FIT): CI 상한이 높으므로, 보조 전략으로 유지

  - Summer→자연경관 (non-FIT): p≈0.08, lift 상한 높음 → 재검토 후보

- 제외할 전략

  - Spring→고궁, Autumn→쇼핑 (non-FIT)